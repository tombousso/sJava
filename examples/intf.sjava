(package examples.intf)

(define-class A () interface)

(define-class B () interface)

(define-class K (A) interface)

(define-class L (A) interface)

(define-class M (A B) interface)

(define-class N (A B) interface)

(define-class O (N) interface)

(define-class Main () public
	((main args String[]) void public static
		(println
			(include
				`,(concat
					(type
						`(begin
							(define k K)
							(define l L)
							(if true k l)
						)
					)
					"\n"
					(type
						`(begin
							(define m M)
							(define o O)
							(if true m o)
						)
					)
					"\n"
					(type
						`(begin
							(if true (as CharSequence "") "")
						)
					)
				)
			)
		)
	)
)
