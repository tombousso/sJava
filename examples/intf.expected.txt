ClassType examples.intf.A
ClassType java.lang.Object
ClassType java.lang.CharSequence
